### ISSUE TITLE
OIDC / OAuth Support

### ISSUE THREAD

**Issue by speatzle** [Ravinou/borgwarehouse#183]
Hi, i recently found BorgWarehouse and have been liking it so far.

I run BorgWarehouse behind Caddy as my Reverse Proxy and am using Authelia for Central User management and Reducing the Attack Surface (only Users logged into Authelia can send HTTP(s) Requests to BorgWarehouse and my other services preventing scanning and potential exploitation).

The only Problem i have is that this means i have to log in twice (once into Authelia and once into BorgWarehouse).

With other services i host i can use OIDC (OAuth) to Login using Authelia's session automatically or by pressing a Dedicated Button on the Login Page.

Alternatively Authelia also Supports Trusted Header SSO where the Service gets HTTP headers from the Proxy containing the Username and Group of the Currently logged in user (in Authelia) completely eliminating the necessity for session management by the Service, here is a introduction: https://www.authelia.com/integration/trusted-header-sso/introduction/

It would be much appreciated if you could support either of these options.

**Comment by rpirsc13** [comment]
I don't believe that Next-Auth has support built for Authelia but a custom OIDC config can be made: https://next-auth.js.org/configuration/providers/oauth#using-a-custom-provider

**Comment by Menschomat** [comment]
This would be really cool.

**Comment by gXkch** [comment]
I would also like to see this. I would contribute myself. Sadly, I am not really familiar with Node / NextJS

**Comment by Ravinou** [comment]
Work has begun and this feature will be included in the next version 3.1 !

**Comment by gXkch** [comment]
Very cool. Thank you, Ravinou.

**Comment by Ravinou** [comment]
After a long evening of work (since I needed to migrate from next-auth to better-auth) I’m pleased to announce the release of version 3.2.0, which now includes support for logging in via OAuth/OIDC.

[Documentation is here !](https://borgwarehouse.com/docs/admin-manual/oauth-oidc/)

Please [support my work](https://liberapay.com/R4VEN), especially if you use borgwarehouse in a professional setting. It always gives me the motivation to keep this wonderful project going.
